lib/queue/producer.ts

```diff
@@ -127,18 +145,68 @@ export async function enqueue(
 	actionRequest: ActionRequestContract,
 	schedule?: ProducerOptionsSchedule,
 ): Promise<ActionRequestContract> {
+	return enqueueWithOptions(
+		logContext,
+		pool,
+		actor,
+		actionRequest,
+		{},
+		schedule,
+	);
+}
+
+/**
+ * @summary Enqueue an action request with options
+ * @function
+ *
+ * @param logContext - log context
+ * @param pool - database connection pool
+ * @param actor - actor id
+ * @param actionRequest - action request contract
+ * @param options : enqueueing options ( see EnqueueOptions )
+ * @param schedule - scheduling options
+ * @returns enqueued action request contract
+ */
+export async function enqueueWithOptions(
+	logContext: LogContext,
+	pool: Pool,
+	actor: string,
+	actionRequest: ActionRequestContract,
+	options: EnqueueOptions,
+	schedule?: ProducerOptionsSchedule,
+): Promise<ActionRequestContract> {
+	// Note that jobNames must be unique for each parameter combination
 	let jobName = 'enqueue-action-request';
 	const jobParameters: string[] = [`'actionRequest'`, '$1'];
 	const values: any[] = [actionRequest];
 
+	let nextParam = 2;
+	if (options.flags) {
+		jobParameters.push('flags := $' + nextParam++);
+		values.push(options.flags);
+		jobName = jobName + '-flags';
+	}
+
+	if (options.priority) {
+		jobParameters.push('priority := $' + nextParam++);
+		values.push(options.priority);
+		jobName = jobName + '-prio';
+	}
+
 	// Handle scheduled actions
 	if (schedule) {
-		jobName = `enqueue-action-request-${uuidv4()}`;
-		jobParameters.push('run_at := $2', 'job_key := $3');
+		jobName = jobName + uuidv4();
+		jobParameters.push(
+			'run_at := $' + nextParam++,
+			'job_key := $' + nextParam++,
+		);
 		values.push(schedule.runAt, schedule.contract);
 	}
 
-	logger.info(logContext, 'Enqueueing action request', {
+	logger.debug(logContext, 'Enqueueing action request', {
```

Comment:
Changing to debug to reduce log noise? It's sometimes useful to have this log in production when something goes wrong.

Reply:
Yes, was logging too much and I haven't found a way ( is there one? ) to filter by category.
Putting back to `info`